### DISCUSSION TITLE
Using edgedb client for node

### DISCUSSION THREAD

**Discussion by haikyuu** [geldata/gel#2838] [Q&A] [answered]
I'm trying to use edgedb in a js project and running
```
	const conn = await edgedb.connect("edgedb://edgedb@localhost/edgedb");
```
hangs forever (no errors and no promise resolved). I'm on a fresh install of edgedb and I created the movies schema from the quick start

**Comment by 1st1** [comment]
Are you using `edgedb project`?

**Reply by haikyuu** [reply]
yes

**Reply by haikyuu** [reply]
Here is my edgedb.toml file
```
[edgedb]
server-version = "1-beta3"

```

**Reply by 1st1** [reply]
Can you try `const conn = await edgedb.connect();`?

**Reply by haikyuu** [reply]
yes, it worked

**Reply by 1st1** [reply]
Awesome.

> angs forever (no errors and no promise resolved). 

This is a bug though, we'll take a look today. It should have failed and suggested the solution.

**Reply by haikyuu** [reply]
thank you

**Comment by haikyuu** [comment, marked as answer]
When using edgedb project, just calling `edgedb.connect()` is enough. 
> This is a bug though, we'll take a look today. It should have failed and suggested the solution.

**Reply by 1st1** [reply]
Yeah, read this blog post by @colinhacks https://www.edgedb.com/blog/introducing-edgedb-projects, it explains what's happening here.

**Comment by 1st1** [comment]
I'm trying to replicate this; the `connect()` call doesn't appear to hang forever, but it does try to connect to the specified instance for a while, until it timeouts:

```
/home/yury/dev/tmp/tmp/node_modules/edgedb/dist/src/client.js:648
                        err = new errors.ClientConnectionFailedTemporarilyError(e.message);
                              ^

ClientConnectionFailedTemporarilyError: connect ECONNREFUSED 127.0.0.1:5656
    at Function.connectWithTimeout (/home/yury/dev/tmp/tmp/node_modules/edgedb/dist/src/client.js:648:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async InnerConnection.reconnect (/home/yury/dev/tmp/tmp/node_modules/edgedb/dist/src/client.js:362:39)
    at async Function.connect (/home/yury/dev/tmp/tmp/node_modules/edgedb/dist/src/client.js:327:9)
    at async main (file:///home/yury/dev/tmp/tmp/index.js:4:12)
    at async file:///home/yury/dev/tmp/tmp/index.js:10:1 {
  source: Error: connect ECONNREFUSED 127.0.0.1:5656
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1142:16) {
    errno: -111,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 5656
  }
}
```

The timeout in question is `waitUntilAvailable`, which is set by default to 30 seconds.

@elprans @tailhook should we lower the default to a lesser value? 

@haikyuu How did you interrupt the script when it was hanging? Did you ^C it? I'm wondering if we can intercept that and print a message explaining what the `connect()` call was actually doing when it was interrupted.

**Reply by 1st1** [reply]
OK, here's a PR to improve the situation https://github.com/edgedb/edgedb-js/pull/147

**Reply by haikyuu** [reply]
Yes, I ^C it.
